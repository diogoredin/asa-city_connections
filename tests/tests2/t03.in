1000
1000
226 751
2 752
763 532
701 604
174 984
239 124
621 18
911 752
324 436
235 131
724 670
173 205
13 593
612 180
312 43
231 463
592 770
617 248
171 17
241 620
192 502
920 349
279 756
308 260
581 488
982 129
958 124
859 721
428 850
91 704
975 99
68 812
858 352
569 94
541 292
273 253
362 676
38 273
928 398
112 277
41 79
980 940
413 784
44 492
844 261
812 311
938 56
488 835
49 928
670 493
138 54
317 208
493 224
754 831
252 190
757 430
677 462
390 103
930 329
895 321
517 78
357 272
445 652
643 333
898 131
66 865
942 683
377 813
433 616
770 882
543 412
229 867
455 150
183 720
438 131
745 988
350 562
398 452
79 207
738 590
81 538
82 959
945 906
648 494
680 294
86 269
87 60
88 534
89 773
692 220
340 203
897 392
526 439
483 392
691 538
96 828
372 293
900 645
228 84
126 870
622 567
294 93
40 858
117 513
550 194
264 822
22 541
521 977
37 361
660 496
726 590
78 998
244 722
635 611
203 539
815 918
534 355
118 654
814 528
666 245
104 650
494 698
321 680
356 868
188 170
16 132
478 520
330 307
809 898
970 257
791 473
132 703
985 464
730 982
282 820
964 859
922 145
184 75
83 930
140 575
832 366
981 304
866 446
194 737
607 326
495 214
651 928
998 458
828 520
923 919
30 398
152 527
35 816
136 593
469 11
758 262
939 618
486 482
780 368
3 533
491 473
335 848
755 888
168 737
212 386
108 722
825 365
567 151
991 992
170 116
248 39
522 539
12 530
55 485
21 922
826 274
778 178
732 873
135 406
736 957
304 852
934 397
887 54
952 713
794 355
663 125
187 273
479 460
676 996
513 494
704 255
242 633
236 390
686 323
195 219
556 837
886 676
371 420
836 858
200 49
98 505
172 28
325 856
213 25
205 108
71 469
771 637
582 798
197 26
92 192
361 291
889 697
302 509
387 199
343 743
659 476
977 983
218 306
133 470
217 957
331 752
391 390
531 640
292 539
979 147
159 193
42 695
901 280
872 74
439 970
265 578
232 481
518 744
381 252
15 328
334 125
56 620
687 886
275 38
240 148
20 416
392 931
598 807
490 410
25 450
246 495
346 245
626 790
310 192
414 826
723 439
605 91
852 758
165 506
925 603
311 350
249 516
258 318
402 521
260 547
803 521
520 226
263 200
199 815
389 123
795 65
386 285
337 151
326 418
555 76
806 169
504 214
46 589
638 190
420 703
474 599
976 47
45 207
564 406
987 634
281 354
186 40
323 28
673 378
285 620
286 674
351 534
664 73
793 469
860 194
404 448
189 792
216 349
446 353
295 550
769 236
297 719
674 793
624 457
883 545
615 56
121 812
39 595
4 178
29 163
141 653
114 250
784 583
929 806
851 150
146 993
169 95
459 82
314 860
262 533
316 970
917 338
119 423
125 928
320 443
535 113
500 985
590 741
707 930
57 310
243 592
873 51
214 564
329 486
959 277
525 889
762 585
26 221
233 262
99 804
515 258
61 59
831 311
764 530
647 163
962 647
406 778
155 285
921 606
854 635
373 562
766 191
884 903
533 166
223 297
27 794
352 330
353 758
566 324
995 452
781 985
256 161
503 774
359 504
634 509
450 551
328 322
363 846
801 672
661 513
366 456
69 744
368 648
720 663
696 665
153 897
679 664
684 577
633 87
924 918
554 263
211 943
821 396
396 980
289 362
234 84
62 350
818 463
384 232
467 865
583 815
456 167
907 955
742 595
105 950
468 226
690 130
694 19
130 863
23 266
574 54
589 299
376 56
388 805
756 837
792 94
298 187
160 76
881 278
158 297
447 950
407 59
885 77
466 763
410 348
181 450
254 971
43 909
287 421
475 162
753 710
417 775
418 414
419 132
206 121
421 924
752 70
300 179
18 780
876 333
644 708
77 27
578 237
562 959
867 681
645 691
913 579
489 978
542 376
1 166
148 641
137 334
548 569
50 928
689 688
630 185
971 521
623 662
14 519
642 57
102 944
399 965
162 41
449 367
667 533
129 481
341 884
134 673
454 593
75 668
746 424
480 229
862 239
472 371
460 305
994 674
179 556
675 661
597 443
856 311
336 448
837 197
364 878
164 836
470 611
471 747
220 440
48 33
824 2
935 189
429 12
190 977
540 786
523 511
558 868
424 468
178 64
662 92
894 684
345 260
551 996
512 757
616 246
101 623
946 383
974 198
492 235
892 215
671 800
774 331
496 225
575 410
787 305
411 441
166 81
965 767
290 879
678 242
6 617
653 583
31 806
224 676
610 537
33 764
912 530
546 735
462 234
536 577
565 317
322 946
725 57
107 261
767 281
497 216
665 447
327 721
865 668
699 847
32 763
914 800
731 789
843 897
620 594
529 225
507 42
586 976
532 470
619 3
374 66
440 908
395 445
890 245
116 191
539 594
432 193
123 794
703 22
175 175
458 826
412 551
845 835
547 150
989 227
383 202
443 37
552 302
819 829
553 243
827 760
904 301
505 372
628 213
747 711
560 129
853 137
728 405
530 533
380 592
782 160
514 216
127 663
157 216
829 245
669 753
568 578
749 288
435 446
573 260
333 604
508 836
19 89
259 462
972 479
167 889
874 339
53 532
208 778
215 667
584 191
944 283
739 838
587 51
950 182
528 184
637 488
591 854
857 9
966 130
293 586
9 348
451 828
303 767
5 869
599 742
600 342
741 508
602 371
457 574
453 590
986 6
430 549
338 17
847 174
280 888
810 192
593 636
193 446
59 221
963 192
557 979
347 895
247 967
618 807
805 395
608 843
785 667
891 885
992 678
319 650
625 714
463 43
627 71
261 728
268 391
538 123
594 187
632 895
705 538
710 292
367 475
266 747
685 457
379 155
631 825
448 320
245 769
65 564
58 308
145 775
588 925
646 157
426 244
161 217
649 109
650 784
180 624
177 676
202 386
654 874
270 668
702 502
822 592
882 725
576 958
442 50
775 621
864 404
715 397
953 641
427 24
639 323
63 19
668 946
961 492
332 4
348 953
7 112
143 763
251 577
527 601
8 876
919 519
257 885
128 765
915 503
681 147
154 434
250 251
60 203
349 645
163 258
910 80
688 645
813 10
238 441
833 948
744 844
693 468
393 203
695 24
358 943
697 519
394 326
714 381
461 395
570 117
139 269
283 327
967 280
437 663
499 590
937 886
708 624
709 848
510 276
711 250
682 122
713 473
811 342
415 943
716 534
201 945
147 561
719 716
369 801
17 227
722 328
926 864
940 739
830 184
955 665
481 700
561 138
698 781
740 975
111 792
267 306
733 327
509 741
444 753
579 738
577 16
729 814
100 612
951 283
601 201
748 447
743 39
718 877
150 774
288 375
595 450
798 493
571 467
276 401
278 974
151 778
115 592
871 865
1000 214
777 246
76 668
477 236
221 587
611 221
761 993
838 268
401 129
34 382
67 678
482 198
109 939
768 670
296 206
36 218
878 976
144 335
773 867
90 169
365 117
993 446
73 443
655 92
779 47
978 95
375 618
933 906
783 259
549 161
434 73
786 858
727 186
788 105
765 608
501 185
997 515
772 353
603 288
142 782
999 198
796 529
225 458
927 341
299 314
277 289
47 370
802 716
823 992
683 158
498 324
28 304
807 726
808 633
759 848
425 474
511 987
905 20
422 509
848 813
94 196
10 168
850 804
52 763
306 223
820 181
452 147
760 193
735 873
405 986
817 704
176 358
24 264
149 308
799 697
385 200
124 198
476 229
204 39
834 377
835 919
506 883
307 783
641 951
948 586
196 707
983 19
706 528
524 809
95 354
700 591
846 182
131 186
11 644
849 386
210 429
839 967
990 815
191 371
841 870
604 115
465 686
596 615
103 74
64 489
484 572
284 447
797 18
750 265
656 621
120 19
776 625
606 395
868 468
869 950
870 475
219 625
789 155
272 248
382 330
875 612
640 606
877 667
182 844
879 738
880 890
291 586
658 223
227 941
305 765
408 270
378 665
537 427
888 241
712 830
318 203
397 241
253 568
893 985
559 543
360 162
717 970
185 33
516 695
899 148
93 566
816 223
403 324
609 939
861 897
613 580
906 670
355 709
908 527
909 388
855 526
409 874
313 147
207 237
580 133
400 330
916 173
309 733
918 192
230 410
969 162
563 402
842 617
473 149
370 790
544 197
344 430
85 895
863 39
209 849
97 320
931 88
932 245
840 326
487 308
84 691
902 39
423 776
896 92
672 696
721 147
941 452
72 108
943 411
614 669
122 70
113 39
947 455
436 365
949 485
545 658
354 448
464 329
269 341
954 625
274 492
956 466
198 216
441 96
342 299
960 979
339 762
237 493
222 70
968 72
657 539
74 468
936 395
957 478
255 365
485 698
110 297
416 646
519 861
652 302
585 93
70 918
973 305
734 338
737 418
502 586
106 382
751 3
800 652
984 51
431 598
636 698
790 173
988 133
804 680
51 136
315 329
572 250
271 531
80 456
156 871
996 201
54 67
903 203
629 602
301 258
1000
14 430 354
212 32 303
673 903 344
673 43 305
935 481 790
254 319 535
474 748 256
86 708 401
361 514 194
130 891 968
130 997 913
860 997 193
300 996 90
803 9 395
818 336 629
651 653 429
13 175 340
582 719 949
413 729 915
866 732 35
443 986 894
952 977 31
952 821 30
331 906 95
923 339 947
48 333 161
731 360 627
827 816 907
892 47 736
565 28 731
728 58 251
212 338 987
261 225 312
981 419 481
942 796 760
348 879 765
104 97 691
558 588 307
228 358 67
865 547 965
764 841 996
878 846 73
514 722 987
486 109 985
187 843 377
16 723 881
669 999 438
387 604 324
328 201 163
690 963 352
560 602 83
397 54 656
801 514 36
341 336 136
330 181 447
317 540 373
27 1000 722
192 115 506
418 587 717
814 332 162
295 640 254
643 260 820
752 123 826
924 355 545
69 807 457
79 533 808
518 485 774
529 629 588
631 276 401
20 614 769
995 743 706
583 943 285
978 997 288
252 879 268
264 734 737
642 319 486
868 790 539
987 877 938
771 769 734
323 783 237
177 524 950
225 425 627
129 616 206
493 331 984
361 353 287
99 60 980
232 884 764
621 760 552
900 466 880
533 179 170
873 155 785
14 874 427
742 896 959
328 529 795
211 100 842
493 803 288
21 51 971
168 599 818
763 705 913
766 60 340
825 156 485
911 452 124
46 839 869
664 519 658
969 932 495
769 266 585
689 142 801
574 341 887
805 773 873
769 568 879
478 393 947
701 145 354
648 80 763
455 656 787
952 283 787
991 349 836
171 132 54
691 929 709
995 156 318
171 82 471
386 740 759
138 422 214
138 886 187
46 107 952
604 318 737
217 683 156
514 100 835
726 150 441
626 253 607
554 403 979
449 270 338
977 175 175
118 780 974
719 984 183
601 999 618
925 888 930
258 498 646
59 693 945
365 70 781
328 874 949
83 60 112
83 159 884
42 334 556
586 151 548
562 272 221
98 199 879
549 880 688
247 893 848
359 642 311
556 314 629
16 487 281
881 5 507
566 408 159
326 773 211
547 909 304
958 144 862
609 737 348
231 802 441
368 185 204
79 121 761
77 150 795
396 529 37
575 810 874
925 251 397
33 274 699
977 719 306
605 32 626
391 552 43
682 90 493
48 616 899
526 327 321
647 437 485
844 515 141
132 530 529
530 969 428
959 223 317
479 296 99
809 632 357
304 352 298
593 210 495
489 440 188
443 268 590
350 974 700
243 903 134
832 367 631
303 870 89
228 597 143
754 750 803
679 419 491
858 547 63
295 544 320
135 48 566
927 936 548
456 899 977
413 467 251
565 632 726
287 451 233
845 823 926
118 360 615
578 805 805
895 943 281
371 827 335
371 840 4
959 466 635
852 185 645
562 434 634
818 604 739
958 728 186
413 559 951
646 863 253
500 576 555
933 85 363
774 912 203
944 180 49
61 637 238
935 106 141
739 993 367
10 431 621
889 914 712
907 960 136
206 699 821
675 853 387
324 205 683
887 560 964
901 299 36
521 740 54
435 577 293
57 818 925
136 481 584
87 207 499
6 124 662
376 417 827
515 300 969
335 919 564
127 999 135
770 288 173
286 628 288
469 456 388
989 891 160
700 968 559
463 250 950
252 625 940
828 463 124
508 817 118
690 631 340
923 813 413
4 268 187
130 106 709
697 431 942
57 697 294
220 957 533
488 218 796
133 24 785
92 605 725
327 299 554
905 382 169
965 725 881
264 536 483
168 54 951
686 14 713
275 819 132
965 905 340
466 824 188
203 406 644
575 927 903
521 712 366
851 637 284
665 145 713
781 360 895
630 484 230
630 649 558
806 631 465
154 848 978
60 848 536
488 647 767
351 742 260
766 36 73
132 366 753
413 254 597
188 714 930
950 319 884
390 950 765
917 562 551
761 559 993
895 856 227
158 306 658
219 943 333
298 631 959
32 293 307
676 442 802
44 862 691
243 161 64
821 616 560
597 115 180
954 315 977
158 164 24
812 227 418
44 938 256
925 961 140
215 64 954
39 473 466
482 993 516
867 161 268
234 849 767
290 379 449
288 996 92
304 718 916
388 613 290
144 850 485
699 430 387
529 878 356
340 685 866
741 84 353
162 277 503
266 954 541
78 655 13
391 775 863
763 861 404
292 466 291
458 208 395
8 149 155
281 175 534
417 993 519
6 888 824
847 306 443
529 875 489
50 878 851
775 779 272
43 539 957
357 465 367
175 869 501
179 101 925
688 697 620
41 585 581
108 510 572
788 93 922
461 960 903
493 926 504
345 777 1
67 737 961
915 156 301
513 719 261
708 278 148
660 418 293
730 354 803
913 671 664
929 613 627
170 248 293
591 721 227
125 729 242
780 685 938
27 484 32
15 600 340
631 342 188
792 204 834
69 947 791
3 207 50
930 894 316
302 368 541
330 256 342
333 871 778
449 272 798
518 568 288
438 430 11
681 375 7
683 284 574
758 877 564
819 973 3
66 652 172
203 785 336
228 103 393
919 84 136
152 268 529
617 179 498
483 334 602
806 752 743
406 269 54
214 992 51
104 316 790
605 223 80
756 946 540
212 240 665
930 6 482
897 765 984
4 667 988
2 582 486
832 704 143
769 301 197
367 161 192
119 443 57
486 221 160
757 233 780
495 562 417
273 446 445
754 24 206
157 559 418
841 909 687
187 645 670
645 111 968
829 749 9
814 707 261
40 579 549
175 509 641
83 787 959
141 176 866
256 468 114
389 18 229
343 243 395
395 485 476
262 450 999
338 499 338
390 251 45
708 718 551
493 651 583
520 752 948
37 783 870
836 775 377
329 742 424
341 201 709
418 408 243
57 639 343
390 53 336
329 675 346
757 855 893
304 7 350
37 285 382
52 382 836
282 834 1000
376 33 321
974 85 706
232 154 957
598 509 257
692 309 18
770 976 1
129 904 364
187 167 110
32 507 350
795 821 977
738 568 539
298 652 414
248 901 512
457 579 297
152 695 361
437 313 817
133 838 442
808 511 603
829 790 768
321 45 137
239 521 15
980 690 741
235 677 174
970 368 935
153 727 859
317 123 722
623 804 907
723 876 449
913 711 588
541 943 235
594 990 691
38 382 466
166 34 264
512 899 329
962 291 1000
170 84 932
647 586 408
420 498 752
404 128 161
883 830 24
16 811 452
324 459 560
598 696 710
459 932 802
247 305 896
62 728 898
542 250 791
542 678 152
256 472 129
320 544 459
91 615 787
330 890 317
699 760 590
632 423 718
864 777 943
78 997 903
670 654 958
945 739 393
982 49 937
362 611 385
229 755 210
193 613 914
541 372 686
168 837 694
483 904 143
835 870 361
359 957 689
160 215 44
21 389 759
983 227 696
470 700 903
173 591 973
213 318 370
140 260 735
504 564 151
312 112 465
184 624 127
174 148 803
668 385 87
27 516 84
275 733 563
40 289 380
146 588 247
491 39 228
946 947 785
25 995 405
439 181 652
312 62 101
44 553 776
442 863 314
346 546 847
752 668 331
332 841 90
637 198 89
859 703 123
911 195 354
287 399 30
314 715 97
565 715 552
920 644 979
726 596 642
534 795 567
766 65 662
597 76 81
940 100 627
981 949 376
71 437 727
511 835 982
197 475 365
618 506 338
33 191 699
527 655 903
15 177 835
539 904 493
825 424 934
312 794 797
159 496 604
671 747 434
792 656 5
558 800 764
898 480 289
964 594 243
247 538 614
474 379 411
592 338 166
752 571 410
948 423 572
214 131 452
141 525 971
763 940 711
747 313 893
325 721 635
720 632 459
521 414 377
238 393 826
29 444 634
770 98 482
860 700 730
169 43 556
45 936 436
637 270 577
279 936 749
275 908 851
23 75 703
312 216 599
236 893 632
486 415 174
963 833 500
977 116 447
496 863 968
507 545 630
626 640 886
627 931 429
490 575 860
928 154 500
723 628 556
295 190 240
229 370 808
920 692 486
724 202 112
322 951 736
812 570 93
942 93 423
675 464 515
975 34 983
373 838 226
784 359 56
325 565 159
429 919 231
582 263 659
170 698 383
188 140 800
663 542 959
196 342 477
482 790 396
548 849 255
974 432 451
433 731 561
692 831 248
363 681 642
638 947 124
320 881 193
419 575 11
724 153 343
290 834 393
894 620 56
183 472 570
892 966 875
659 113 89
762 139 130
680 281 755
694 148 848
13 517 210
19 109 859
533 619 684
331 711 280
770 280 744
793 313 295
666 282 907
486 245 204
939 108 341
886 11 883
522 300 528
365 899 369
117 586 841
31 947 105
543 716 659
402 28 345
328 72 296
890 834 698
17 765 462
732 804 569
798 604 760
685 709 810
398 862 833
628 544 72
981 136 727
375 880 587
413 921 633
102 274 370
153 999 640
235 422 334
162 570 108
445 441 425
659 69 590
845 434 158
873 166 507
922 335 796
249 191 287
595 572 255
38 537 545
35 395 667
391 782 609
543 571 621
265 189 448
939 280 926
859 900 314
616 725 495
687 728 902
189 528 449
914 291 785
539 502 832
757 836 969
356 208 321
249 347 102
938 584 31
567 275 701
494 749 487
108 403 868
303 10 880
987 786 614
96 93 882
569 524 344
165 586 730
745 403 990
387 634 618
56 711 570
648 47 764
96 718 187
372 857 814
998 956 440
919 306 254
554 807 675
88 148 838
633 817 72
41 36 108
380 219 98
883 566 165
994 703 79
551 492 389
892 810 305
535 904 110
739 400 410
451 711 633
998 309 959
797 54 402
219 54 291
885 344 878
12 933 251
359 842 727
392 956 812
396 162 507
645 904 408
747 672 31
39 602 39
142 307 698
665 743 270
815 669 264
684 845 826
200 596 700
665 477 651
300 378 35
934 414 468
987 315 571
215 839 987
720 955 177
188 886 500
872 605 145
372 302 118
327 782 564
297 594 745
214 110 351
555 344 710
410 510 681
595 727 646
171 849 839
858 710 553
38 393 720
443 277 14
387 446 476
203 607 70
894 506 54
182 816 862
75 222 258
910 315 489
186 785 675
686 137 710
356 296 681
904 822 335
389 576 282
87 107 108
377 345 321
828 971 643
907 912 627
7 385 516
389 874 854
985 862 293
612 477 861
134 471 212
125 214 400
454 481 4
775 305 769
940 672 374
460 843 817
983 284 649
45 953 662
648 802 940
923 717 698
362 151 952
489 524 38
536 198 558
453 370 760
788 227 665
894 658 546
653 593 6
551 355 657
598 523 493
308 46 36
96 791 675
662 671 950
104 803 840
458 820 989
479 448 487
244 956 782
892 457 987
568 538 27
229 375 505
129 332 611
224 17 175
771 105 599
610 576 15
972 559 260
479 475 858
364 482 400
390 674 79
497 266 508
270 776 356
223 151 592
532 967 34
830 1000 642
744 1000 449
615 374 369
440 808 862
464 984 862
367 122 712
4 383 388
644 48 854
726 181 261
35 657 534
200 95 141
815 981 199
866 816 689
357 573 729
226 453 359
659 407 620
809 258 953
184 352 321
562 546 298
390 941 391
521 854 97
496 944 703
758 641 963
447 640 464
918 271 970
833 906 650
617 510 653
294 588 523
504 609 486
63 1000 327
45 678 236
622 34 545
101 918 440
424 452 438
981 29 185
203 679 434
39 693 188
704 17 695
101 960 597
843 608 749
88 841 563
959 735 125
91 242 336
333 441 563
102 425 681
19 257 947
826 773 303
581 89 635
435 498 473
927 487 642
565 882 555
319 163 143
188 772 610
775 789 945
826 122 770
76 817 3
437 339 404
720 181 375
801 427 75
109 315 195
557 606 592
632 354 209
522 769 567
226 243 20
148 528 276
828 702 338
437 149 381
445 615 630
979 296 162
340 864 953
55 562 879
898 186 743
340 375 13
691 931 423
401 839 458
30 277 230
801 501 437
1 689 620
194 797 975
188 25 734
432 349 776
822 951 316
567 937 535
262 559 301
938 682 631
660 713 670
56 27 112
241 636 20
827 426 872
739 147 765
155 143 226
3 182 155
630 746 272
724 358 99
489 561 328
239 339 830
837 267 38
475 462 63
320 590 217
522 589 300
456 568 723
114 834 849
733 85 1
635 807 512
258 767 161
661 631 24
174 557 362
778 734 258
496 123 662
913 768 745
917 74 511
562 927 116
472 650 182
540 73 822
187 26 418
931 790 834
556 172 91
81 809 140
178 375 523
24 903 128
950 394 383
562 603 975
204 877 742
439 759 28
472 94 661
698 879 224
126 929 781
531 404 68
213 405 159
491 611 777
555 613 441
803 654 205
476 869 248
798 465 135
186 689 68
275 470 869
894 369 340
736 337 8
404 31 864
922 246 259
617 706 935
889 298 946
68 127 500
87 609 107
22 140 695
44 20 573
260 255 971
643 419 316
799 113 783
837 905 634
38 967 843
123 436 798
421 399 957
505 655 711
762 753 501
61 563 758
982 434 764
670 550 52
384 973 175
292 237 830
731 990 448
627 893 777
176 227 640
323 32 16
701 840 326
71 409 22
506 580 192
313 916 496
985 545 148
230 984 324
390 307 568
445 16 720
98 381 330
897 209 651
310 57 720
280 902 623
808 519 130
428 400 753
259 960 467
216 204 108
521 471 470
311 595 595
503 618 83
492 51 162
89 469 134
11 120 653
588 823 88
241 820 439
416 519 801
411 650 669
622 33 541
766 435 447
581 93 12
522 751 900
448 145 19
923 412 938
530 988 513
150 222 433
139 655 329
539 485 982
736 451 266
35 891 12
617 932 677
