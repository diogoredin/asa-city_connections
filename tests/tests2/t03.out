293596
501 499
